### PR TITLE
[ATL][GLOBAL] Enable _DEBUG globally if debugging!!!

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Improve debuggability.
JIRA issue: [CORE-17505](https://jira.reactos.org/browse/CORE-17505)

## Proposed changes

- Define `_DEBUG` globally if debugging!!!
- ATL sees `_DEBUG` and treats it correctly.
- Delete `_DEBUG` hacks.
- Undefine `_DEBUG` in `drivers`, CRT, and `explorer_old`.

## TODO

- [ ] Fix build.
- [ ] Do tests carefully.

### CODE REVIEW

File: sdk/lib/atl/atldef.h [hunk 1]
```diff
@@ -7,6 +7,15 @@
 
 #pragma once
 
+#ifdef NDEBUG
```
Comment (HBelusca) [inline sdk/lib/atl/atldef.h:10]: NDEBUG is a define just to control the printout (to debugger) of DPRINT traces (not the DPRINT1). And "DPRINT(1)" are not used in ATL.

File: sdk/lib/atl/atldef.h [hunk 2]
```diff
@@ -7,6 +7,15 @@
...
+    #undef DBG
```
Comment (HBelusca) [inline sdk/lib/atl/atldef.h:11]: DBG is a systemwide macro that tells whether code is compiled in debug mode or not. Disabling it that way (especially wrt. NDEBUG) is not safe.

**Comment by HBelusca** [comment]
!!! 🤣

**Comment by katahiromz** [comment]
Too buggy. Too dangerous. Closed.

**Pull request closed without merging**